### ISSUE TITLE
Automate release

### ISSUE THREAD

**Issue by priscavdsluis** [Deltares/HYDROLIB-core#451]
We should find out if its possible to automate our release procedure (creating the package, uploading it to PyPi, etc.)
Example: https://github.com/Deltares/hatyan/blob/main/.github/workflows/pypi-upload.yml

**Comment by MRVermeulenDeltares** [comment]
Usefull information: https://packaging.python.org/en/latest/guides/publishing-package-distribution-releases-using-github-actions-ci-cd-workflows/

**Comment by tim-vd-aardweg** [comment]
I created a GitHub workflow that should be able to automate the release steps as listed under https://deltares.github.io/HYDROLIB-core/0.4.1/guides/contributing/#releasing

Once this workflow is merged into main, a new action should be available under the GitHub Actions page. When clicked, you should be able to provide it with a major, minor and patch version for the new release.

The workflow should then:
- checkout the code
- install python, poetry and all required dependencies
- generate the changelog
- bump the version
- push (tag) to git
- build the package
- upload the package to PyPI
- create a new GitHub release for the new version including the correct name, version and changelog.

Once the workflow has been reviewed, we can merge it back to main. We can then test this workflow when we are going to release a new version for hydrolib-core in the near future and make changes if needed.
